packages/graphile-build/src/utils.js

```diff
@@ -32,11 +32,22 @@ export const formatInsideUnderscores = (fn: (input: string) => string) => (
   return `${start}${fn(middle)}${end}`;
 };
 
+export const formatLastSegment = (fn: (input: string) => string) => (
+  str: string
+) => {
+  const matches = str.match(/^([\s\S]*?)([A-Z]?[a-z0-9]*)$/);
```

Comment:
I think this regexp might be vulnerable to ReDoS. An alternative approach might be:

```js
const matches = str.match(/[A-Z_][a-z0-9]*$/);
const prefix = matches ? str.substr(0, matches.index) : "";
const word = matches ? matches[0].replace(/^_/, '') : str;
```